packages/lib/modules/swap/useIsPoolSwapUrl.tsx

```diff
@@ -0,0 +1,6 @@
+import { usePathname } from 'next/navigation'
+
+export function useIsPoolSwapUrl() {
+  const pathname = usePathname()
+  return pathname.includes('/pools') && pathname.includes('/swap')
+}
```

Comment:
I don't think this is a good approach. I was just trying to point out that the logic using the URL might not be the best way to decide if it's a pool swap. Now, if we decide to support pool swaps in the swap UI you'll have to have two conditionals all over the place.

The alternative approach would be to maintain a `isPoolSwap` state variable in the SwapProvider. The default can be set by a useSwap param (i.e. a param passed into the provider). So the swap layout used in the pool context would take a prop like `isPoolSwap` which is then passed into the provider. Then in the future we can easily switch on this flag with UI changes in the swap form.

Reply:
We already have the `isPoolSwap` into the provider. 
But we do need the two conditionals. 

`isPoolSwap` is used to display the specific pool swap parts

`useIsPoolSwapUrl` is used to detect the specific case where the swap is happening within a pool actions layout, meaning that receipt page and redirects work in a different way

Reply:
Ok, maybe let's see.